extensions/ql-vscode/src/config.ts

```diff
@@ -112,12 +113,13 @@ export interface QueryHistoryConfig {
   onDidChangeConfiguration: Event<void>;
 }
 
-const CLI_SETTINGS = [ADDITIONAL_TEST_ARGUMENTS_SETTING, NUMBER_OF_TEST_THREADS_SETTING, NUMBER_OF_THREADS_SETTING];
+const CLI_SETTINGS = [ADDITIONAL_TEST_ARGUMENTS_SETTING, NUMBER_OF_TEST_THREADS_SETTING, NUMBER_OF_THREADS_SETTING, MAX_PATHS];
 
 export interface CliConfig {
   additionalTestArguments: string[];
   numberTestThreads: number;
   numberThreads: number;
+  maxPaths: number;
```

Comment:
I think you can unit test this - take a look at https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/vscode-tests/minimal-workspace/config.test.ts and add it to the list of values. (I think it will go under the CLIConfigListener rather than the QueryServerConfigListener.)

Reply:
I've added it to that test file [here](https://github.com/github/vscode-codeql/pull/931/files#diff-f8a6abadae6a8fd37afb3cd61659401d1f7119e14c31ac26f2abb6cd57d37d55). Silly question, but what should `values` be? I couldn't work out if there's a logic to it 🙃

Reply:
Not a silly question, I wondered the same while looking at it. I think the intent is to test a range of values that we expect to be handled: for this kind of test it's a good idea to check the minimum, the maximum, some common value in the middle, and a zero/empty value. If the test is capable of handling errors, then also test some value that shouldn't be allowed.

Reply:
It looks like the test only lets me test 2 values? https://github.com/github/vscode-codeql/blob/e119218828d5106b0fa2d3547be8208ebeb93938/extensions/ql-vscode/src/vscode-tests/minimal-workspace/config.test.ts#L34-L36

I can't quite work out if the test is doing anything with the values themselves, or just listening for when they change 😕

Reply:
Ohh you are right. The test is just making sure the listeners handle a change. So `[0, 1]` sounds fine.
We can look separately at creating an integration test that produces multiple paths - right now we have none so there's nothing to update.